### PR TITLE
[AAP-17646] Include Workflow Approval List Row tests

### PR DESCRIPTION
**Pull request by fullsushidev**
Relates to [AAP-17646](https://issues.redhat.com/browse/AAP-17646)

- The tests marked with skip will be added to another PR on top of this one. (Total of 3 PR)
- The function `actAssertAndDeleteWorkflowApproval` concentrates the code reused by 3 test cases, making it easier to mantain and understand.

### CODE REVIEW

**Comment by vishsanghishetty** [comment]
@fullsushidev  it looks like the lint check failed
 https://github.com/ansible/ansible-ui/actions/runs/8524638711/job/23349705516?pr=1926
can you please fix them and push the changes? 

`npm run tsc` to see your TS errors before you push. alternatively, you can also do `npm run prepush` to check both TS and formatting errors

**Comment by fullsushidev** [comment]
@vishsanghishetty I pushed my changes with the errors because I need a new pair of eyes to help me fix them. I'm not really familiar with the interfaces.

Also makes me curious why does it work locally even with those errors =/

**Comment by vishsanghishetty** [comment]
> @vishsanghishetty I pushed my changes with the errors because I need a new pair of eyes to help me fix them. I'm not really familiar with the interfaces.
> 
> Also makes me curious why does it work locally even with those errors =/

test can always run successfully locally with or without correct TS interfaces. only when you try to push the checks will fail and stop the code from getting in due to TS errors. these coding standards help keep our code secure. @fullsushidev

File: cypress/e2e/awx/administration/workflowApproval.cy.ts [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+      });
+  });
+
+  it('admin can deny and then delete a workflow approval from the list row item', () => {
+    cy.wait('@launchWFJT')
+      .its('response')
+      .then((response: { body: { workflow_job: number } }) => {
+        cy.getAwxWFApprovalByWorkflowJobID(response.body.workflow_job).then((wfa) => {
+          cy.visit(`/administration/workflow-approvals/`);
+          cy.filterTableByTextFilter('id', wfa.id.toString());
```
Comment (nixocio) [inline cypress/e2e/awx/administration/workflowApproval.cy.ts:150]: I saw that filter is being applied twice. It seems like `getTableRow` also filters by  filterTableByTextFilter

```Cypress.Commands.add(
  'getTableRow',
  (columnDataCy: string, text: string, options?: { disableFilter?: boolean }) => {
    if (options?.disableFilter !== true) {
      cy.filterTableByTextFilter(columnDataCy, text);
    }
    if (columnDataCy === 'id') {
      cy.contains(`[data-cy="row-id-${text}"]`, text).parents('tbody');
    } else {
      cy.contains(`[data-cy="${columnDataCy}-column-cell"]`, text).parents('tr');
    }
  }
);```

File: cypress/e2e/awx/administration/workflowApproval.cy.ts [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+          cy.getTableRow('id', wfa.id.toString()).within(() => {
+            cy.getByDataCy('actions-column-cell').within(() => {
+              cy.getByDataCy('deny').click();
+            });
+          });
```
Comment (nixocio) [inline cypress/e2e/awx/administration/workflowApproval.cy.ts:155]: Where do we assert that in main table the status was updated to `Denied` or `Approved`?

image

File: cypress/support/auth.ts [hunk 3]
```diff
@@ -48,6 +48,49 @@ Cypress.Commands.add('awxLogin', () => {
   cy.visit(`/`, { retryOnStatusCodeFailure: true, retryOnNetworkFailure: true });
 });
 
+Cypress.Commands.add('awxLoginTestUser', (username: string, password: string) => {
```
Comment (nixocio) [inline cypress/support/auth.ts:51]: Is this change related to work flow approvals? This command is not used as part of any test, right?

File: cypress/e2e/awx/administration/workflowApproval.cy.ts [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+    cy.deleteAwxUser(user, { failOnStatusCode: false });
+    cy.deleteAwxUser(userWFApprove, { failOnStatusCode: false });
+    cy.deleteAwxUser(userWFCancel, { failOnStatusCode: false });
+    cy.deleteAwxUser(userWFDeny, { failOnStatusCode: false });
+  });
+
+  function refreshToDelete(wfaID: string) {
+    // FIXME: the page needs to be refreshed to button be available
+    cy.visit(`/administration/workflow-approvals/`);
+    cy.filterTableByTextFilter('id', wfaID);
```
Comment (nixocio) [inline cypress/e2e/awx/administration/workflowApproval.cy.ts:99]: Same comment as below? We could make this a cypress command?

**Review by nixocio — COMMENTED** [review]
hey @fullsushidev overall good job. I left a couple of questions/suggestions.

File: cypress/e2e/awx/administration/workflowApproval.cy.ts [hunk 5]
```diff
@@ -0,0 +1,206 @@
+import { Inventory } from '../../../../frontend/awx/interfaces/Inventory';
+import { InventorySource } from '../../../../frontend/awx/interfaces/InventorySource';
+import { JobTemplate } from '../../../../frontend/awx/interfaces/JobTemplate';
+import { Organization } from '../../../../frontend/awx/interfaces/Organization';
+import { Project } from '../../../../frontend/awx/interfaces/Project';
+import { User } from '../../../../frontend/awx/interfaces/User';
+import { WorkflowJobTemplate } from '../../../../frontend/awx/interfaces/WorkflowJobTemplate';
+import { WorkflowNode } from '../../../../frontend/awx/interfaces/WorkflowNode';
+
+// describe('Workflow Approvals', () => {
```
[the remark this answers is not in the document]
Comment (fullsushidev) [reply cypress/e2e/awx/administration/workflowApproval.cy.ts:10]: yes and no. This structure will hold the incoming tests after this PR.

File: cypress/support/auth.ts (hunk 3 above)
Comment (fullsushidev) [reply cypress/support/auth.ts:51]: This is related to workflow approvals, but I'll leave a comment.
Basically, when we have the access tab ready, some of the tests aim to:
admin assign user to approve workflow > logout > user logs in and approves

File: cypress/e2e/awx/administration/workflowApproval.cy.ts (hunk 4 above)
Comment (fullsushidev) [reply cypress/e2e/awx/administration/workflowApproval.cy.ts:99]: Might be, I want to be sure this will indeed need to be refreshed or if we can find a better way to wait for the workflow status. I also left it inside the test spec, since it only works for workflow approvals.

**Review by akus062381 — APPROVED** [review]
Looks good!

**Review by nixocio — APPROVED** [review]
